src/main/java/servlets/SearchServlet.java

```diff
@@ -47,20 +40,20 @@
 public class SearchServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String desiredLabel = request.getParameter("label");
-    ImmutableList<Receipt> receipts = getReceiptsWithMatchingLabel(desiredLabel);
+    String categories = request.getParameter("categories");
+    ImmutableList<Receipt> receipts = getReceiptsWithMatchingCategories(categories);
 
     Gson gson = new Gson();
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(receipts));
   }
 
-  /** Returns ImmutableList of receipts from datastore with the same label as desiredLabel. */
-  private ImmutableList<Receipt> getReceiptsWithMatchingLabel(String desiredLabel) {
-    // Set filter to retrieve only receipts with label equal to desiredLabel.
-    Filter matchingLabels = new FilterPredicate("label", FilterOperator.EQUAL, desiredLabel);
+  /** Returns ImmutableList of receipts from datastore with the same categories. */
+  private ImmutableList<Receipt> getReceiptsWithMatchingCategories(String categories) {
+    // Set filter to retrieve only receipts with categories equal to categories.
```

Comment:
Does this mean the user has to remember the exact set of categories they originally entered or that was extracted from the receipt? I would have assumed that if we had the following:

receipt1 - category1, category2, category3
receipt2 - category2
receipt3 - category3
receipt4 - category4

searching for category1 would return receipt1 and searching for category2 would return receipt1 and receipt2. If you searched for category3, category4, the search would return receipt1, receipt3, and receipt4 so would be an OR of the categories the user is searching for.

Reply:
Searching for only one category would return the right receipts. But for the MVP we agreed that the user would only be able to search one category for now